pkg/server/tls_windows.go

```diff
@@ -0,0 +1,342 @@
+			cn.winMut.Lock()
+			defer cn.winMut.Unlock()
+			cert := &tls.Certificate{
+				Certificate: [][]byte{cn.serverCert.Raw},
+				PrivateKey:  cn.serverSigner,
+				// These seem to the be safest to use, tested on Win10, Server 2016, 2019, 2022
+				SupportedSignatureAlgorithms: []tls.SignatureScheme{
+					tls.PKCS1WithSHA256,
+					tls.PKCS1WithSHA384,
+					tls.PKCS1WithSHA512,
+				},
+			}
+			return cert, nil
+		},
+		// Windows has broad support for 1.2, only 2022 has support for 1.3
+		MaxVersion: tls.VersionTLS12,
+	}
+
+	switch c.ClientAuth {
+	case "RequestClientCert":
+		config.ClientAuth = tls.RequestClientCert
+	case "RequireAnyClientCert", "RequireClientCert": // Preserved for backwards compatibility.
+		config.ClientAuth = tls.RequireAnyClientCert
+	case "VerifyClientCertIfGiven":
+		config.ClientAuth = tls.VerifyClientCertIfGiven
+	case "RequireAndVerifyClientCert":
+		config.ClientAuth = tls.RequireAndVerifyClientCert
+	case "", "NoClientCert":
+		config.ClientAuth = tls.NoClientCert
+	default:
+		return fmt.Errorf("invalid ClientAuth %q", c.ClientAuth)
+	}
+	// Kick off the refresh handler
+	go cn.startUpdateTimer()
+	l.windowsCertHandler = cn
+	l.tlsConfig = config
+	l.cfg = c
+	return nil
+}
+
+// this is the ASN1 Object Identifier for TemplateID
+var asnTemplateOID = "1.3.6.1.4.1.311.21.7"
+
+type winCertStoreHandler struct {
+	cfg          WindowsCertificateFilter
+	subjectRegEx *regexp.Regexp
+
+	winMut       sync.Mutex
+	serverCert   *x509.Certificate
+	serverSigner crypto.PrivateKey
+	// We have to store the identity to access the signer (its an api call), the client does NOT need the signer
+	serverIdentity certstore.Identity
+	clientRootCA   *x509.Certificate
+
+	stopUpdateTimer chan struct{}
+}
+
+func (c *winCertStoreHandler) startUpdateTimer() {
+	if c.cfg.ServerRefreshInterval == 0 {
+		c.cfg.ServerRefreshInterval = 5 * time.Minute
+	}
+
+	select {
+	case <-c.stopUpdateTimer:
+		if c.serverIdentity != nil {
+			c.serverIdentity.Close()
+		}
+		c.serverCert = nil
+		c.serverSigner = nil
+		return
+	case <-time.After(c.cfg.ServerRefreshInterval):
+		err := c.refreshCerts()
+		if err != nil {
+			return
+		}
+
+	}
+}
+
+// refreshCerts is the main work item in certificate store, responsible for finding the right certificate
+func (c *winCertStoreHandler) refreshCerts() error {
+	c.winMut.Lock()
+	defer c.winMut.Unlock()
+	// Close the server identity if already set
+	if c.serverIdentity != nil {
+		c.serverIdentity.Close()
+	}
+	var serverIdentity certstore.Identity
+	var clientIdentity certstore.Identity
+	var err error
+	// This handles closing all our various handles
+	defer func() {
+		// we have to keep the server identity open if we want to use it, BUT only if an error occurred, else we need it
+		// open to sign
+		if serverIdentity != nil && err != nil {
+			serverIdentity.Close()
+		}
+		// Client identity does need to be open since we dont need to sign
```

Comment:
s/does/doesn't?

Reply:
`does NOT` is what I meant to write, good catch =)